### ISSUE TITLE
NVDA 2022.3 beta 1 has wrongly referred the character description of Chinese character

### ISSUE THREAD

**Issue by LittleStar-VIP** [nvaccess/nvda#13978]
### Steps to reproduce:
1. type the Chinese character "香" into somewhere like a notepad
2. make sure the nvda locale is set to tw
3. pressthe numpad 2 twice to hear the character description
4. the generated character description is "香 香港 香蕉 香煙 蚊香 香味 香料 香水 芳香 檀香 中國香港 在香港以外"
But according to the tw character description file the character description for the character 香 should be
香 香水 香港 香味 芳香 百里香 焚香頂禮
 
### Expected behavior:
nvda should refer to the character description dic file of the respective chinese locale
#### NVDA installed/portable/running from source:
portable
#### NVDA version:
nvda 2022.3 beta 1
#### Windows version:
windows home ed 21h2
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
this problem does not exist in previous nvda, e.g. nvda 2022.2
#### If NVDA add-ons are disabled, is your problem still occurring?
not using any addon
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
not tried

**Comment by LittleStar-VIP** [comment]
Would other Chinese nvda users please test out this

**Comment by seanbudd** [comment]
Could you please reproduce this issue and provide a [log file](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps) of the behaviour. Ensure your log level is set to debug in general preferences.

It would be great to have a log of 2022.2 performing the right behaviour and 2022.3beta1 performing the wrong behaviour using the same sample.

**Comment by LittleStar-VIP** [comment]
Oh sorry I found out the reason.
A member of the tw mailing list member reminded me that the option "Trust voice's language when processing characters and symbols" is checked by default.
So when I was doing the test, I was using a hong kong cantonese voice to do the test, and so the character descriptionthat was spoken by nvda was the character description of hong kong locale.
After unchecking the option, Trust voice's language when processing characters and symbols, I found that both the nvda 2022.3 beta 1 and the pr are working fine as expected.
